lib/Dialect/FIRRTL/NLATable.cpp

```diff
@@ -37,6 +37,34 @@ NLATable::NLATable(Operation *operation) {
   }
 }
 
+void NLATable::insert(NonLocalAnchor nla) {
+  symToOp[nla.sym_nameAttr()] = nla;
+  for (auto ent : nla.namepath()) {
+    if (auto mod = ent.dyn_cast<FlatSymbolRefAttr>())
+      nodeMap[mod.getAttr()].push_back(nla);
+    else if (auto inr = ent.dyn_cast<hw::InnerRefAttr>())
+      nodeMap[inr.getModule()].push_back(nla);
+  }
+}
+
+void NLATable::erase(NonLocalAnchor nla) {
+  symToOp.erase(nla.sym_nameAttr());
+  auto removeNLAfromList = [&](SmallVector<NonLocalAnchor, 4> &nlaList) {
+    for (auto it = nlaList.begin(), e = nlaList.end(); it != e; ++it) {
+      if (*it != nla)
+        continue;
+      nlaList.erase(it);
+      break;
+    }
+  };
+  for (auto ent : nla.namepath()) {
+    if (auto mod = ent.dyn_cast<FlatSymbolRefAttr>()) {
+      removeNLAfromList(nodeMap[mod.getAttr()]);
+    } else if (auto inr = ent.dyn_cast<hw::InnerRefAttr>())
+      removeNLAfromList(nodeMap[inr.getModule()]);
+  }
```

Comment:
Extra `{` `}` on the `if`.

I think you can use `llvm::erase_value(nodeMap[mod.getAttr()], nla)` instead of `removeNLAfromList`